src/extension.ts

```diff
@@ -116,16 +121,19 @@ export async function _stopExistingDfaRun(context: vscode.ExtensionContext, outp
 	if (pid) {
 		try {
 			process.kill(pid as number, SIGKILL);
+			void context.workspaceState.update(Constants.WORKSPACE_DFA_PROCESS, undefined);
 			await vscode.window.showInformationMessage(messages.graphEngine.dfaRunStopped);
 		} catch (e) {
+			// Exception is thrown by process.kill if between the time the pid exists and kill is executed, the process
+			// ends by itself. Ideally it should clear the cache, but doing this as an abundant of caution.
+			void context.workspaceState.update(Constants.WORKSPACE_DFA_PROCESS, undefined);
```

Comment:
Since this is in both branches, does it make sense to put it in a finally block?

Reply:
I want the cache update to happen before the information message is shown and as soon as the process is killed. The reason is to try and avoid any race condition possible. I once hit an issue where the process was all killed and the cache was not cleared and this made starting another DFA run impossible (unless the project is reloaded). I made these small changes of moving the code around to clear the cache and haven't seen any issues since then.